engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -75,66 +88,13 @@ class PartwiseJoin : public kernel {
 	std::unique_ptr<TableSchema> left_schema{nullptr};
  	std::unique_ptr<TableSchema> right_schema{nullptr};
 
-	std::unique_ptr<ral::frame::BlazingTable> load_set(BatchSequence & input, std::shared_ptr<ral::cache::CacheMachine> cache, bool load_all){
-		std::size_t bytes_loaded = 0;
-		std::vector<std::unique_ptr<ral::frame::BlazingTable>> tables_loaded;
-		if (input.wait_for_next()){
-			tables_loaded.emplace_back(input.next());
-			bytes_loaded += tables_loaded.back()->sizeInBytes();
-		} else {
-			return nullptr;
-		}
-		// NOTE this used to be like this:
-		// while ((input.has_next_now() && bytes_loaded < join_partition_size_theshold) ||
-		//  	(load_all && input.wait_for_next())) {
-		// with the idea that it would just start processing as soon as there was data to process.
-		// This actually does not make it faster, because it makes it so that there are more chunks to do pairwise joins and therefore more join operations
-		// We may want to revisit or rethink this
-
-		while ((input.wait_for_next() && bytes_loaded < join_partition_size_theshold) ||
-					(load_all && input.wait_for_next())) {
-			tables_loaded.emplace_back(std::move(input.next()));
-			bytes_loaded += tables_loaded.back()->sizeInBytes();
-
-			//If the concatenation so far produces a string overflow, we put back the batch because in this case the order of the elements does not matter
-			if(!load_all && ral::utilities::checkIfConcatenatingStringsWillOverflow(tables_loaded)){
```

Comment:
where does this part happen now

Reply:
This is now handled by the concatenating cache. This here was effectively doing the same as a concatenating cache. I made it so that now the inputs to a join are concatenating caches. This way, we have less redundant code and also a join can operate on a more traditional, simpler pullFromCache, instead of all this logic